docs/recipes/Mocking-API-Calls.md

```diff
@@ -2,35 +2,121 @@
+   The `someApiUrl` above is setup within the `appConfig` of the module. Check out the [One App configurations guide to learn how to set this up](../modules/App-Configuration.md#provideStateConfig).
 
-The `set-dev-endpoints` command links your module's dev endpoints file to One App.
+2. Configure the Parrot scenarios. When modules are generated using [One App module generator](https://github.com/americanexpress/one-app-cli/tree/master/packages/generator-one-app-module) a mock folder is created, this contains sample scenarios.
 
-A `dev.endpoints.js` file contains all the information One App needs to configure [one-app-dev-proxy](https://github.com/americanexpress/one-app-dev-proxy)
-(our reverse proxy and mocking server that runs during development) and can be used to set remote endpoints
-for your Module to use during local development.
+   > 🐦 Parrot is a set of tools that allow you to create HTTP mocks and organize them into scenarios in order to develop your app against different sets of data
 
-```bash
-$ npm run set-dev-endpoints ../[module-name]/dev.endpoints.js
-```
+    ```
+    module
+    ├── README.md
+    ├── mock
+    |   └── scenarios.js
+    |
+    ├── package.json
+    └── src
+        └── index.js
+    ```
 
-A `dev.endpoints.js` file looks like:
+   Update the file to accommodate all the scenarios that you'd like to mock using Parrot. Refer to [Parrot scenarios](https://github.com/americanexpress/parrot/blob/master/SCENARIOS.md) to learn how you can setup different scenarios.
+
+3. Set up the `dev-middleware.js` and `dev.endpoints.js`.
+  
+   - `dev-middleware.js`
+  
+   A `dev-middleware.js` file allows you to setup a custom middleware configuration. This file is created by default when [One App module generator](https://github.com/americanexpress/one-app-cli/tree/master/packages/generator-one-app-module) is used in generating the module. Ensure that [`parrot-middleware`](https://github.com/americanexpress/parrot/tree/master/packages/parrot-middleware) is installed.
+
+   **Running within One App.**
+
+   Use the `set-middleware` command to link your module's custom dev middleware file to One App
+
+    ```bash
+    npm run set-middleware [path-to-your-module]/dev.middleware.js
+    ```
+
+   **Running with One App runner**
+
+    When using [one-app-runner](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-runner), this can be setup within the package.json as follows:
+  
+    ```json
+    "one-amex": {
+      "runner": {
+        "parrotMiddleware": "[path-to-your-module]/dev.middleware.js"
+      },
+    }
+    ```
+
+    The file `dev-middleware.js` contains the following parrot configurations.
+
+    ```javascript
+    const parrot = require('parrot-middleware');
+    const scenarios = require('./mock/scenarios');
+
+    module.exports = (app) => app.use(parrot(scenarios));
+    ```
+
+   - `set-dev-endpoints.js`
```

Comment:
I think this needs to be a heading rather than a bullet-point so it is more visible

Reply:
Update this let me know what you thin